packages/public-api/src/web.ts

```diff
@@ -0,0 +1,52 @@
+export interface MaskNetworkAPIs {
+    web_echo<T>(arg: T): Promise<T>
+    /**
+     * @returns A fully quantified URL in forms of "holoflows-extension://...." or "moz-extension://...." (based on the platform)
+     */
+    getDashboardURL(): Promise<string>
+
+    /**
+     * @deprecated Use settings_getLaunchPageSettings()
+     */
+    getSettings(key: 'launchPageSettings'): Promise<LaunchPage>
+
+    /**
+     * @returns A stringified JSON string.
+     * @example [[{ network: "twitter.com", connected: false } ]]
+     */
+    getConnectedPersonas(): Promise<string>
+    app_isPluginEnabled(pluginID: string): Promise<boolean>
+    app_setPluginStatus(pluginID: string, enabled: boolean): Promise<void>
+    setting_getNetworkTraderProvider(network: NetworkType): Promise<TradeProvider | undefined>
+    setting_setNetworkTraderProvider(network: NetworkType, provider: TradeProvider): Promise<void>
+    settings_getTrendingDataSource(): Promise<DataProvider>
+    settings_setTrendingDataSource(provider: DataProvider): Promise<void>
+    settings_getLaunchPageSettings(): Promise<LaunchPage>
+}
+export enum LaunchPage {
+    facebook = 'facebook',
+    twitter = 'twitter',
+    dashboard = 'dashboard',
+}
+// This type MUST be sync with NetworkType in packages/web3-shared/src/types/index.ts
+export enum NetworkType {
+    Ethereum = 'Ethereum',
+    Binance = 'Binance',
+    Polygon = 'Polygon',
+    Arbitrum = 'Arbitrum',
+}
+export enum DataProvider {
```

Comment:
IMO, we should not move plugin features into another public shared package (The previous one is `@masknet/web3-shared`). It brings unclear division. We need a plugin package for this.

Reply:
Types in this package (`public-api`) is a promise to the external developer (iOS/Android/...), which mean it must be self-contained, and the inner content must not be somewhere else that might be changed without being aware of changing a public API that might break someone.